cognite/neat/rules/models/_rules/dms_architect_rules.py

```diff
@@ -0,0 +1,115 @@
+import abc
+from datetime import datetime
+from typing import Any, ClassVar
+
+from cognite.client.data_classes.data_modeling import PropertyType
+from pydantic import Field, field_validator
+
+from cognite.neat.rules.models._rules.information_rules import InformationMetadata
+
+from .base import BaseMetadata, BaseRules, RoleTypes, SheetEntity, SheetList
+from .domain_rules import DomainMetadata
+
+subclasses = list(PropertyType.__subclasses__())
+_PropertyType_by_name: dict[str, type[PropertyType]] = {}
+for subclass in subclasses:
+    subclasses.extend(subclass.__subclasses__())
+    if abc.ABC in subclass.__bases__:
+        continue
+    try:
+        _PropertyType_by_name[subclass._type.casefold()] = subclass
+    except AttributeError:
+        ...
+del subclasses  # cleanup namespace
+
+
+class DMSArchitectMetadata(BaseMetadata):
+    role: ClassVar[RoleTypes] = RoleTypes.dms_architect
+    space: str
+    external_id: str = Field(alias="externalId")
+    version: str | None = Field(
```

Comment:
Attach filed validator from information architect here.

Reply:
Moved out a type for it.